spacy/tokens/doc.pyx

```diff
@@ -820,6 +820,69 @@ cdef class Doc:
             set_children_from_heads(self.c, length)
         return self
 
+    @staticmethod
+    def from_docs(docs, space_delimiter=True, attributes=None):
+        """Concatenate multiple Doc objects to form a new one.
+
+        docs (list): A list of Doc objects.
+        space_delimiter (bool): Put spaces between the docs (i.e. one space for each pair of end
+            and start token of subsequent docs).
+        attributes (list): Optional list of attribute ID ints or attribute name strings.
+        RETURNS (Doc): The new doc that is containing the other docs, or None if no doc was given.
+
+        DOCS: https://spacy.io/api/doc#from_docs
+        """
+        if not docs:
+            return None
+
+        vocab = {doc.vocab for doc in docs}
+        if len(vocab) > 1:
+            raise AssertionError(Errors.E189)
```

Comment:
ok, I see. But it is an error of incompatibility of `Vocab` objects, isn't it? That is more of a theoretical question, since I get that it is generally not useful.